harper-core/Cargo.toml

```diff
@@ -24,6 +24,7 @@ thiserror = "2.0.9"
 unicode-blocks = "0.1.9"
 unicode-width = "0.2.0"
 levenshtein_automata = { version = "0.2.1", features = ["fst_automaton"] }
+typst-syntax = "0.12.0"
```

Comment:
Typst is pretty big and completely unused for certain applications (i.e. Obsidian). Would you mind putting Typst support behind a feature flag?

Reply:
Alternatively, you could make the Typst parser its own crate.

Reply:
How would making it a seperate crate help? It would still be a dependency...

I feel like feature flag is the way to go. I assume we would have it enabled for harper-ls and harper-cli?